getNodeByUnit/index.ts

```diff
@@ -0,0 +1,56 @@
+import { AzureFunction, Context, HttpRequest } from '@azure/functions';
+import { Node } from '../shared/node/node.interface';
+import { User, Role } from '../shared/user/user.interface';
+import { httpErrorHandler, ForbiddenFunctionError, ValidationFunctionError } from '../shared/error/error';
+import HttpTriggerResult from '../shared/httpTriggerResult';
+import NodeModel from '../shared/node/node.model';
+import getConnection from '../shared/mongoose';
+import authWrapper from '../shared/authWrapper';
+
+const functionName = 'getNodeByUnit';
+
+const getNodeByUnit: AzureFunction = async (_context: Context, req: HttpRequest, user: User): Promise<HttpTriggerResult> => {
+    try {
+        const allowedRoles = [Role.RamadIturAssistant, Role.RamadIturOfUnit, Role.Technical, Role.Itur, Role.Mada];
+        const nodesGroupId = req.query.nodesGroupId;
+        const unitId = req.query.unitId;
+        const nodesGroupIdToSearch = nodesGroupId === '' ? undefined : nodesGroupId;
+        const unitIdToSearch =
+            user.role === Role.RamadIturAssistant || user.role === Role.RamadIturOfUnit ? user.unitId : unitId === '' ? undefined : unitId;
+        const searchingFilter: any = { unitId: unitIdToSearch };
```

Comment:
```suggestion
        const searchingFilter: any = { unitId: unitIdToSearch, nodeGroupId: nodeGroupIdToSearch };
```
When the role is `RamadIturAssistant`, The `nodeGroupId` won't be included in the filter.